### PR TITLE
First checkin of disable and enable flags for cadc-access-control-admin

### PR DESCRIPTION
**Pull request by hjeeves**
Includes changes in cadc-access-control-server to support this.

### CODE REVIEW

File: cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java [hunk 1]
```diff
@@ -69,6 +69,7 @@
 
  package ca.nrc.cadc.ac.admin;
 
+import com.sun.prism.Texture;
```
Comment (brianmajor) [inline cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java:72]: This must be an error... a graphics library or something?

File: cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/AbstractUserCommand.java [hunk 2]
```diff
@@ -137,7 +137,7 @@ protected void printUser(final User user)
             this.systemOut.println();
             if (user.personalDetails != null)
             {
-                this.systemOut.println(user.personalDetails.toStringFormatted());
+                // Returns users that are locked (disabled or marked deleted)
```
Comment (brianmajor) [inline cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/AbstractUserCommand.java:140]: This looks like it must be a mistake too.  Doesn't that println needs to be in there for output to show?

File: cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java [hunk 3]
```diff
@@ -247,6 +248,28 @@ protected boolean isValid(final ArgumentMap am) throws UsageException
             count++;
     	}
 
+        userID = am.getValue("disable");
+        if (userID != null	)
```
Comment (brianmajor) [inline cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java:252]: Looks like a tab is in this line

File: cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java [hunk 4]
```diff
@@ -247,6 +248,28 @@ protected boolean isValid(final ArgumentMap am) throws UsageException
...
+            if (this.hasValue(userID))
+            {
+                this.command = new DisableUser(userID);
+            }
+
+            count++;
+        }
+
+        userID = am.getValue("enable");
+        if (userID != null	)
```
Comment (brianmajor) [inline cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java:263]: Looks like a tab is in this line

File: cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java [hunk 5]
```diff
@@ -312,6 +335,7 @@ public static String getUsage()
     	sb.append("--approve=<userid> --dn=<dn> : Approve user with userid=<userid> and set the\n");
     	sb.append("                             : distinguished name to <dn>\n");
     	sb.append("--reject=<userid>            : Delete this user request\n");
+        sb.append("--disable=<userid>           : Disable this user account\n");
```
Comment (brianmajor) [inline cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java:338]: Missing the 'enable' in usage?

File: cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/AbstractUserCommand.java (hunk 2 above)
Comment (brianmajor) [reply cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/AbstractUserCommand.java:140]: Or maybe it was moved elsewhere?

File: cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/ViewUser.java [hunk 6]
```diff
@@ -105,12 +105,25 @@ protected void execute()
...
+                this.systemOut.println("User Status: PENDING");
+            }
+            catch (AccessControlException | UserNotFoundException ue)
+            {
+                // Check to see if it's locked
+                log.debug("principal: " + this.getPrincipal());
+                User user = this.getUserPersistence().getLockedUser(this.getPrincipal());
+                this.printUser(user);
+                this.systemOut.println("User Status: LOCKED");
+            }
```
Comment (brianmajor) [inline cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/ViewUser.java:126]: The status will be a nice addition

**Review by brianmajor — CHANGES_REQUESTED** [review]

File: cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java (hunk 1 above)
Comment (hjeeves) [reply cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java:72]: Yeah - quick-typing include? Weird... Gone now.

File: cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/AbstractUserCommand.java (hunk 2 above)
Comment (hjeeves) [reply cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/AbstractUserCommand.java:140]: I think this was a ball drop - I'd meant to remove the comment.  Weird that I didn't catch it in testing, though...

Comment (hjeeves) [reply cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/AbstractUserCommand.java]: Added back in, can see personal details using --view now.

File: cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java (hunk 3 above)
Comment (hjeeves) [reply cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java]: looks fixed in diff now.

File: cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java (hunk 4 above)
Comment (hjeeves) [reply cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java]: looks fixed in diff now.

File: cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java (hunk 5 above)
Comment (hjeeves) [reply cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/CmdLineParser.java:338]: added in now

File: cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/ViewUser.java (hunk 6 above)
Comment (hjeeves) [reply cadc-access-control-admin/src/main/java/ca/nrc/cadc/ac/admin/ViewUser.java:126]: I found that both active and pending accounts were displayed in --view, and maybe after practice one could discern between it. The other reason for this statement is to allow the admin to tell if an account is locked, or truly doesn't exist in the system. (the add of .getLockedUser() ) - before if a user was locked it was never displayed with --view.

**Review by brianmajor — APPROVED** [review]

**Pull request merged**